eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -370,6 +431,595 @@ def test_update_user_in_tenant_user_not_found(self, param: str, value: str) -> N
+        enrollment_data = {
+            param: user_data[param],
+            "course_id": "course-v1:OpenedX+DemoX+NonExistentCourse",
+            "mode": self.mode,
+        }
+
+        response = self.create_enrollment(self.tenant_x, enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("non_field_errors", response_data)
+        self.assertEqual(response_data["non_field_errors"], ["Course not found"])
+
+    @ddt.data("email", "username")
+    def test_create_valid_user_mode_invalid_course_for_tenant(self, param: str) -> None:
+        """
+        Test creating a enrollment with a valid user, valid mode, and a course from another tenant.
+
+        Open edX definitions tested:
+        - `check_edxapp_enrollment_is_valid`
+        - `validate_org`
+
+        Expected result:
+        - The status code is 400.
+        - The response contains an error message about the course not found on the tenant.
+        """
+        user_data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_y, user_data)
+        enrollment_data = {
+            param: user_data[param],
+            "course_id": self.course_id,
+            "mode": self.mode,
+        }
+
+        response = self.create_enrollment(self.tenant_y, enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("course_id", response_data)
+        self.assertEqual(response_data["course_id"], [f"Invalid course_id {self.course_id}"])
+
+    @ddt.data("email", "username")
+    def test_create_valid_user_course_invalid_mode(self, param: str) -> None:
+        """
+        Test creating a enrollment with a valid user, valid course, and a not available mode in a tenant.
+
+        Open edX definitions tested:
+        - `check_edxapp_enrollment_is_valid`
+        - `api.validate_course_mode`
+
+        Expected result:
+        - The status code is 400.
+        - The response contains an error message about the mode not found.
+        """
+        user_data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, user_data)
+        enrollment_data = {
+            param: user_data[param],
+            "course_id": self.course_id,
+            "mode": "masters",
+        }
+
+        response = self.create_enrollment(self.tenant_x, enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("non_field_errors", response_data)
+        self.assertEqual(response_data["non_field_errors"], ["Mode not found"])
+
+    @ddt.data("email", "username")
+    def test_force_create_valid_user_course_mode_not_allowed(self, param: str) -> None:
+        """
+        Test creating a enrollment with a valid user, valid course, and a not available mode in a tenant.
```

Comment:
```suggestion
        Test creating an enrollment with a valid user, valid course, and an unavailable mode in a tenant.
```

Reply:
What's the difference between this test case and `test_create_valid_user_course_invalid_mode`?

Reply:
When we use `force=True` the enrollment is created even if the mode is unavailable in the course.

Reply:
Can we clarify that in the docstring?

Reply:
Done: https://github.com/eduNEXT/eox-core/commit/490a19eee770c832882cb730a705843911370df9